pkg/util/deepcopy/deepcopy_test.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2021 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deepcopy_test
+
+import (
+	"testing"
+
+	"github.com/go-test/deep"
+
+	"k8c.io/kubermatic/v2/pkg/util/deepcopy"
+)
+
+func TestStringInterfaceMapCopy(t *testing.T) {
+
+	testcases := []struct {
+		name           string
+		copyFrom       map[string]interface{}
+		copyTo         map[string]interface{}
+		expectedResult map[string]interface{}
+		expectedErr    error
+	}{
+		{
+			name: "scenario 1: deep copy one map to another",
+			copyFrom: map[string]interface{}{
+				"key":      []string{"values", "values"},
+				"otherKey": bob{name: "Bob"},
+			},
+			copyTo: map[string]interface{}{},
+			expectedResult: map[string]interface{}{
+				"key":      []string{"values", "values"},
+				"otherKey": bob{name: "Bob"},
```

Comment:
Nit: check out the comment below:
```suggestion
				"otherKey": struct { name string }{ name: "bob" },
```